pkg/integrations/redis_exporter/redis_exporter_test.go

```diff
@@ -0,0 +1,165 @@
+package redis_exporter //nolint:golint
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/prometheus/pkg/textparse"
+	"github.com/stretchr/testify/require"
+)
+
+const addr string = "localhost:6379"
+
+type testCase struct {
+	name                   string
+	cfg                    Config
+	expectedMetrics        []string
+	expectConstructorError bool
+	envVars                map[string]string
+}
+
+var testCases = []testCase{
```

Comment:
Small nit: test tables should be defined inline in the test that uses them with the following pattern:

```
tt := []struct{
  name string 
  cfg Config 
  expectedMetrics []string 
  expectedConstructorError bool 
  envVars map[string]string 
}{
  // ... 
}

// test case in test table 
for _, tc := range tt {
  t.Run(tc.name, func(t *testing.T) {
     // run test case 
  })
}
```

Reply:
nice, I assume this avoids test runs being cached, even though we've changed the test input.

Reply:
Nah, there's no functional difference in doing it one way over the other (but subtests can be filtered out at runtime to focus on a specific test case). 

The rest of the codebase defines the test cases inside the test function, so I just want to be consistent with that.